Examples/Tests/ohm_solver_EM_modes/analysis_rz.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+#
+# --- Analysis script for the hybrid-PIC example producing EM modes.
+
+import dill
+import matplotlib
+from matplotlib import colors
+import matplotlib.pyplot as plt
+import numpy as np
+from openpmd_viewer import OpenPMDTimeSeries
+import scipy.fft as fft
+from scipy.interpolate import RegularGridInterpolator
+from scipy.special import j1, jn, jn_zeros
+
+from pywarpx import picmi
+
+constants = picmi.constants
+
+# load simulation parameters
+with open(f'sim_parameters.dpkl', 'rb') as f:
+    sim = dill.load(f)
+
+diag_dir = "diags/field_diags"
+
+ts = OpenPMDTimeSeries(diag_dir, check_all_files=True)
+
+def transform_spatially(data_for_transform):
+    # interpolate from regular r-grid to special r-grid
+    interp = RegularGridInterpolator(
+        (info.z, info.r), data_for_transform,
+        method='linear'
+    )
+    data_interp = interp((zg, rg))
+
+    # Applying manual hankel in r
+    # Fmz = np.sum(proj*data_for_transform, axis=(2,3))
+    Fmz = np.einsum('ijkl,kl->ij', proj, data_interp)
+    # Standard fourier in z
+    Fmn = fft.fftshift(fft.fft(Fmz, axis=1), axes=1)
+    return Fmn
+
+def process(it):
+    print(f"Processing iteration {it}", flush=True)
+    field, info = ts.get_field('E', 'y', iteration=it)
+    F_k = transform_spatially(field)
+    return F_k
+
+# grab the first iteration to get the grids
+Bz, info = ts.get_field('B', 'z', iteration=0)
+
+nr = len(info.r)
+nz = len(info.z)
+
+nkr = 12 # number of radial modes to solve for
+
+r_max = np.max(info.r)
+
+# create r-grid with points spaced out according to zeros of the Bessel function
+r_grid = jn_zeros(1, nr) / jn_zeros(1, nr)[-1] * r_max
+
+zg, rg = np.meshgrid(info.z, r_grid)
+
+# Setup Hankel Transform
+j_1M = jn_zeros(1, nr)[-1]
+r_modes = np.arange(nkr)
+
+A = (
+    4.0 * np.pi * r_max**2 / j_1M**2
+    * j1(np.outer(jn_zeros(1, max(r_modes)+1)[r_modes], jn_zeros(1, nr)) / j_1M)
+    / jn(2 ,jn_zeros(1, nr))**2
+)
+
+# No transformation for z
+B = np.identity(nz)
+
+# combine projection arrays
+proj = np.einsum('ab,cd->acbd', A, B)
```

Comment:
It is just a trick for doing the Hankel transform for all z-slices through a single matrix multiplication, instead of multiplying by `A` inside a for-loop over `nz`. 
Basically, `A` has shape `m` x `nr` (where `m` is the number of radial modes we want to keep), while the field data has shape `nr x nz`. So I create an identity matrix of shape `n x nz` (where `n` is the number of axial modes - in this case `n = nz`), and then reorder the axis so that `proj` has shape `m x n x nr x nz` so that I can do a one shot transform from the field data array.
This technique is nice if you only care about a small number of axial modes as well since you can do the Hankel and Fourier transform in a single multiplication (setting up `B` appropriately).